_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+      "driver_info": {
+        "username": "admin",
+        "password": "password",
+        "port": "6230",
+        "address": "ipmi://192.168.111.1:6230",
+        "deploy_kernel": "http://172.22.0.1/images/ironic-python-agent.kernel",
+        "deploy_ramdisk": "http://172.22.0.1/images/ironic-python-agent.initramfs"
+      },
+      "ports": [
+        {
+          "address": "00:00:e0:4b:24:8b",
+          "pxe_enabled": true
+        }
+      ],
+      "properties": {
+        "local_gb": "50",
+        "cpu_arch": "x86_64"
+      }
+    },
+    {
+      "name": "node-1",
+      "driver": "ipmi",
+      "resource_class": "baremetal",
+      "driver_info": {
+        "username": "admin",
+        "password": "password",
+        "port": "6231",
+        "address": "ipmi://192.168.111.1:6231",
+        "deploy_kernel": "http://172.22.0.1/images/ironic-python-agent.kernel",
+        "deploy_ramdisk": "http://172.22.0.1/images/ironic-python-agent.initramfs"
+      },
+      "ports": [
+        {
+          "address": "00:00:e0:4b:24:8f",
+          "pxe_enabled": true
+        }
+      ],
+      "properties": {
+        "local_gb": "50",
+        "cpu_arch": "x86_64"
+      }
+    },
+...
+...
+...
+```
+
+<br>
+
+> **Information:** This role is also used to tear down the virtual infrastructure depending on the variable [libvirt_action](https://github.com/metal3-io/metal3-dev-env/blob/2b5d8e76f33d143757d1f0b9b1e82dc662245b9c/vm-setup/roles/libvirt/defaults/main.yml#L2) inside the Ansible role: `setup or teardown`.
+
+<br>
+
+> * **Virtbmc**
+
+> This role is only executed if the bare metal virtual machines are created in libvirt, because `vbmc` needs libvirt to emulate a real BMC.
+<br>
+<br>
+> **Information:**  VirtualBMC (`vmbc`) tool simulates a Baseboard Management Controller (BMC) by exposing IPMI responder to the network and talking to libvirt at the host vBMC is running at. Basically, manipulate virtual machines which pretend to be bare metal servers.
+<br>
+<br>
+> The `virtbmc` Ansible role create the `vbmc` and `sushy-tools` configuration in the host for each virtual bare metal nodes. Note that each virtual bare metal host will have a different `vbmc` socket exposed in the host. The communication to each `vbmc` is needed by the `BareMetal operator` to start, stop, configure the boot order, etc during the provisioning stage. Finally, this folders containing the configuration will be mounted by the `vbmc` and  `sushy-tools` containers.
+<br>
+<br>
+
+> ```sh
+[alosadag@eko1 metal3-dev-env]$ sudo ls -l --color /opt/metal3-dev-env/virtualbmc
+total 0
+drwxr-x---. 2 root root 21 Feb  5 11:07 sushy-tools
+drwxr-x---. 4 root root 70 Feb  5 11:08 vbmc
+```
+
+Next, both host provisioning and baremetal interfaces are configured. Provisioning interface, as the name suggests, it will be used to provision the virtual bare metal hosts by means of the `BareMetal operator`. This interface is configured with an static IP (172.22.0.1):
```

Comment:
The `provisioning` interface, as the name suggests, will be used... (i.e. add _The_, remove _it_)

Reply:
Also: Bare Metal Operator (if you changed that above)